src/nnet3/nnet-chain-example.cc

```diff
@@ -221,6 +221,14 @@ static void MergeSupervision(
                    &output_supervision);
   output->supervision.Swap(&output_supervision);
 
+  int32 example_stride = 0;
+  for (auto &index: inputs[0]->indexes)
```

Comment:
please check that this is a C++11 construct.

Reply:
It's a feature since C++11: https://en.cppreference.com/w/cpp/language/range-for.
I have also compiled it on CLSP grid with -std=c++11.